### PR TITLE
[NETBEANS-1673]:Added support for auto-complete of enum case values i…

### PR DESCRIPTION
**Pull request by arusinha**
…n switch expression(jdk-12 feature)

Jira link : https://issues.apache.org/jira/browse/NETBEANS-1673

### CODE REVIEW

File: java/java.source.base/src/org/netbeans/api/java/source/ReflectionHelper.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+     *
+     * @param className the fully qualified name of the desired class.
+     * @param methodName the name of the method
+     * @param parameterTypes the method parameter types list
+     * @param obj the object the underlying method is invoked from
+     * @param paramArgs the arguments used for the method call
+     * @return the result of the invoked method.
+     * @since 2.41
+     */
+    public static Object invokeMethod(String className, String methodName, List<Class> parameterTypes, Object obj, Object... paramArgs) {
```
Comment (jlahoda) [inline java/java.source.base/src/org/netbeans/api/java/source/ReflectionHelper.java]: Sorry, but I don't think we should have a method like this in the API. That is not bringing any real value to the clients, and the use is (hopefully) transient anyway.

My recommendation is to isolate the reflection into the TreeShims, providing "high-level" methods (like getExpressions(Tree), enforcing that the Tree is a SwitchExpressionTree). With:
https://github.com/apache/incubator-netbeans/pull/1074

The only copy of TreeShims is in java.source.base, and the methods can be reused by java.hints and java.completion (and if needed by any other module).

File: java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java [hunk 2]
```diff
@@ -49,6 +49,8 @@
 import org.netbeans.api.java.source.support.ErrorAwareTreePathScanner;
 import org.netbeans.api.java.source.support.ReferencesCount;
 import org.netbeans.api.lexer.TokenSequence;
+
```
Comment (jlahoda) [inline java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java]: Nit: unnecessary empty lines.

File: java/java.source.base/src/org/netbeans/api/java/source/ReflectionHelper.java (hunk 1 above)
Comment (arusinha) [reply java/java.source.base/src/org/netbeans/api/java/source/ReflectionHelper.java]: Had removed ReflectionHelper in recent commit instead used TreeShims class

File: java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java (hunk 2 above)
Comment (arusinha) [reply java/java.completion/src/org/netbeans/modules/java/completion/JavaCompletionTask.java]: formaating corrected in recent commit

**Comment by arusinha** [comment]
> I don't think there should be an API change for this, and esp. not a method that only invokes reflection.
> 
> Also, tests appear to be missing?
Had removed the API is cecent commit
Test Cases will be added as part of below task once nb-javac for jdk-12 is available
https://issues.apache.org/jira/browse/NETBEANS-1964

**Pull request merged**